src/js/node/http/Server.hx

```diff
@@ -37,107 +37,168 @@ import js.Error;
+	#else
 	var Connect:ServerEvent<IncomingMessage->Socket->Buffer->Void> = "connect";
+	#end
 
 	/**
-		Emitted each time a client requests a http upgrade.
+		This event is emitted when a new TCP stream is established.
+		`socket` is typically an object of type net.Socket. Usually users will not want to access this event.
+		In particular, the socket will not emit `'readable'` events because of how the protocol parser attaches to the socket.
+		The `socket` can also be accessed at `request.connection`.
 
-		If this event isn't listened for, then clients requesting an upgrade will have their connections closed.
+		This event can also be explicitly emitted by users to inject connections into the HTTP server. In that case,
+		any `Duplex` stream can be passed.
 
-		Listener arguments:
-			request - arguments for the http request, as it is in the request event
-			socket - network socket between the server and client
-			head - the first packet of the tunneling stream, this may be empty
+		If `socket.setTimeout()` is called here, the timeout will be replaced with `server.keepAliveTimeout`
+		when the socket has served a request (if `server.keepAliveTimeout` is non-zero).
+	**/
+	var Connection:ServerEvent<Socket->Void> = "connection";
 
-		After this event is emitted, the request's socket will not have a data event listener,
-		meaning you will need to bind to it in order to handle data sent to the server on that socket.
+	/**
+		Emitted each time there is a request.
+		There may be multiple requests per connection (in the case of HTTP Keep-Alive connections).
 	**/
-	var Upgrade:ServerEvent<IncomingMessage->Socket->Buffer->Void> = "upgrade";
+	#if haxe4
+	var Request:ServerEvent<(request:IncomingMessage, response:ServerResponse) -> Void> = "request";
+	#else
+	var Request:ServerEvent<IncomingMessage->ServerResponse->Void> = "request";
+	#end
 
 	/**
-		If a client connection emits an 'error' event - it will forwarded here.
+		Emitted each time a client requests an HTTP upgrade.
+		Listening to this event is optional and clients cannot insist on a protocol change.
+
+		After this event is emitted, the request's socket will not have a `'data'` event listener,
+		meaning it will need to be bound in order to handle data sent to the server on that socket.
 	**/
-	var ClientError:ServerEvent<Error->Socket->Void> = "clientError";
+	#if haxe4
+	var Upgrade:ServerEvent<(request:IncomingMessage, socket:Socket, buffer:Buffer) -> Void> = "upgrade";
+	#else
+	var Upgrade:ServerEvent<IncomingMessage->Socket->Buffer->Void> = "upgrade";
+	#end
 }
 
 /**
-	HTTP server
+	This class inherits `from net.Server`.
 **/
 @:jsRequire("http", "Server")
 extern class Server extends js.node.net.Server {
 	/**
-		Limits maximum incoming headers count, equal to 1000 by default.
-		If set to 0 - no limit will be applied.
+			Limit the amount of time the parser will wait to receive the complete HTTP headers.
```

Comment:
Looks like there's too much indentation here?

I think running formatting it with haxe-formatter should take care of that - I started using it on hxnodejs in a recent commit.

Reply:
Thanks! I forgot to run the formatter.

Reply:
That reminds me, I should put formatOnSave into the VSCode config of this repo. :)